### ISSUE TITLE
Memory leak in factory_create_extension

### ISSUE THREAD

**Issue by jchapma** [389ds/389-ds-base#5729]
**Issue Description**
The functionality to extend an operation object introduces a memory leak

**Package Version and Platform:**
 - Package and version:  389-Directory/2.3.2 (main branch)

**Steps to Reproduce**
1. Run the sync repl cookie add del CI tests - LSAN_OPTIONS=exitcode=0:log_threads=1:verbosity=1 PYINSTALL=1 py.test -sv dirsrvtests/tests/suites/syncrepl_plugin/basic_test.py::test_sync_repl_cookie_add_del
2. Check asan output
```
Indirect leak of 64 byte(s) in 1 object(s) allocated from:
    #0 0x7f3c1aa8f7f7 in __interceptor_calloc (/lib64/libasan.so.6+0xb07f7)
    #1 0x7f3c1a26d29d in slapi_ch_calloc ldap/servers/slapd/ch_malloc.c:175
    #2 0x7f3c1a2c1716 in factory_create_extension ldap/servers/slapd/factory.c:320
    #3 0x437c09 in connection_add_operation ldap/servers/slapd/connection.c:2086
    #4 0x4371ae in connection_activity ldap/servers/slapd/connection.c:1941
    #5 0x444b93 in handle_pr_read_ready ldap/servers/slapd/daemon.c:1621
    #6 0x442ede in ct_list_thread ldap/servers/slapd/daemon.c:1306
    #7 0x7f3c19442553 in _pt_root ../../.././nspr/pr/src/pthreads/ptthread.c:201
    #8 0x7f3c193d4431 in start_thread /usr/src/debug/glibc-2.31-74-gd0c84d22b6/nptl/pthread_create.c:477
    #9 0x7f3c19cae6d2 in __clone (/lib64/libc.so.6+0x1016d2)

Indirect leak of 40 byte(s) in 1 object(s) allocated from:
    #0 0x7f3c1aa8f7f7 in __interceptor_calloc (/lib64/libasan.so.6+0xb07f7)
    #1 0x7f3c1a26d29d in slapi_ch_calloc ldap/servers/slapd/ch_malloc.c:175
    #2 0x7f3c1a34f43f in op_stat_constructor ldap/servers/slapd/operation.c:690
    #3 0x7f3c1a2c18ac in factory_create_extension ldap/servers/slapd/factory.c:328
    #4 0x437c09 in connection_add_operation ldap/servers/slapd/connection.c:2086
    #5 0x4371ae in connection_activity ldap/servers/slapd/connection.c:1941
    #6 0x444b93 in handle_pr_read_ready ldap/servers/slapd/daemon.c:1621
    #7 0x442ede in ct_list_thread ldap/servers/slapd/daemon.c:1306
    #8 0x7f3c19442553 in _pt_root ../../.././nspr/pr/src/pthreads/ptthread.c:201
    #9 0x7f3c193d4431 in start_thread /usr/src/debug/glibc-2.31-74-gd0c84d22b6/nptl/pthread_create.c:477
    #10 0x7f3c19cae6d2 in __clone (/lib64/libc.so.6+0x1016d2)

Indirect leak of 24 byte(s) in 1 object(s) allocated from:
    #0 0x7f3c1aa8f7f7 in __interceptor_calloc (/lib64/libasan.so.6+0xb07f7)
    #1 0x7f3c1a26d29d in slapi_ch_calloc ldap/servers/slapd/ch_malloc.c:175
    #2 0x7f3c11fe968f in supplier_operation_extension_constructor ldap/servers/plugins/replication/repl_opext.c:25
    #3 0x7f3c1a2c18ac in factory_create_extension ldap/servers/slapd/factory.c:328
    #4 0x437c09 in connection_add_operation ldap/servers/slapd/connection.c:2086
    #5 0x4371ae in connection_activity ldap/servers/slapd/connection.c:1941
    #6 0x444b93 in handle_pr_read_ready ldap/servers/slapd/daemon.c:1621
    #7 0x442ede in ct_list_thread ldap/servers/slapd/daemon.c:1306
    #8 0x7f3c19442553 in _pt_root ../../.././nspr/pr/src/pthreads/ptthread.c:201
    #9 0x7f3c193d4431 in start_thread /usr/src/debug/glibc-2.31-74-gd0c84d22b6/nptl/pthread_create.c:477
    #10 0x7f3c19cae6d2 in __clone (/lib64/libc.so.6+0x1016d2)

Indirect leak of 16 byte(s) in 1 object(s) allocated from:
    #0 0x7f3c1aa8f7f7 in __interceptor_calloc (/lib64/libasan.so.6+0xb07f7)
    #1 0x7f3c1a26d29d in slapi_ch_calloc ldap/servers/slapd/ch_malloc.c:175
    #2 0x7f3c11fe98b5 in consumer_operation_extension_constructor ldap/servers/plugins/replication/repl_opext.c:54
    #3 0x7f3c1a2c18ac in factory_create_extension ldap/servers/slapd/factory.c:328
    #4 0x437c09 in connection_add_operation ldap/servers/slapd/connection.c:2086
    #5 0x4371ae in connection_activity ldap/servers/slapd/connection.c:1941
    #6 0x444b93 in handle_pr_read_ready ldap/servers/slapd/daemon.c:1621
    #7 0x442ede in ct_list_thread ldap/servers/slapd/daemon.c:1306
    #8 0x7f3c19442553 in _pt_root ../../.././nspr/pr/src/pthreads/ptthread.c:201
    #9 0x7f3c193d4431 in start_thread /usr/src/debug/glibc-2.31-74-gd0c84d22b6/nptl/pthread_create.c:477
    #10 0x7f3c19cae6d2 in __clone (/lib64/libc.so.6+0x1016d2)

Indirect leak of 8 byte(s) in 1 object(s) allocated from:
    #0 0x7f3c1aa8f7f7 in __interceptor_calloc (/lib64/libasan.so.6+0xb07f7)
    #1 0x7f3c1a26d29d in slapi_ch_calloc ldap/servers/slapd/ch_malloc.c:175
    #2 0x7f3c1a34f42c in op_stat_constructor ldap/servers/slapd/operation.c:689
    #3 0x7f3c1a2c18ac in factory_create_extension ldap/servers/slapd/factory.c:328
    #4 0x437c09 in connection_add_operation ldap/servers/slapd/connection.c:2086
    #5 0x4371ae in connection_activity ldap/servers/slapd/connection.c:1941
    #6 0x444b93 in handle_pr_read_ready ldap/servers/slapd/daemon.c:1621
    #7 0x442ede in ct_list_thread ldap/servers/slapd/daemon.c:1306
    #8 0x7f3c19442553 in _pt_root ../../.././nspr/pr/src/pthreads/ptthread.c:201
    #9 0x7f3c193d4431 in start_thread /usr/src/debug/glibc-2.31-74-gd0c84d22b6/nptl/pthread_create.c:477
    #10 0x7f3c19cae6d2 in __clone (/lib64/libc.so.6+0x1016d2)
```

**Additional context**
This issue is a subset of https://github.com/389ds/389-ds-base/issues/5646

**Comment by jchapma** [comment]
edb43569f..8ac0c5d11  389-ds-base-2.0 -> 389-ds-base-2.0
2c35f7309..b1bdf5021  389-ds-base-2.1 -> 389-ds-base-2.1
143ea6530..3588838fb  389-ds-base-2.2 -> 389-ds-base-2.2
ca41bf590..8be8c46f6  389-ds-base-2.3 -> 389-ds-base-2.3

**Comment by jchapma** [comment]
Merged into main and cherry picked back to 2.0 - 2.3
